conu/backend/docker/image.py

```diff
@@ -221,6 +226,99 @@ def push(self, repository=None, tag=None):
                                         self.name, error)
         return image
 
+    def set_transport(self, transport=None, path=None):
+        """
+        :param transport: from where will be this image copied
+        :param path in filesystem
+        :return: self
+        """
+        if not transport:
+            return self
+        if self.transport == transport and self.path == path:
+            return self
+        path_required = [Transport.DIRECTORY, Transport.DOCKER_ARCHIVE, Transport.OCI]
+        if transport in path_required:
+            self.path = self.mount(path).mount_point
+        elif transport == Transport.OSTREE:
+            if path and not os.path.isabs(path):
+                raise ConuException("Path '", path, "' for OSTree transport is not absolute")
+            self.path = path
+        else:
+            self.path = None
+
+        self.transport = transport
+        return self
```

Comment:
I would expect that `set_*` method does not return anything.

Reply:
I used it as a shortcut when creating new images.. I guess that putting transport and path as optional arguments in contructor instead makes more sense.

Reply:
and if I did so.. I think if I should raise error/warning if something fishy happens.. For example someone would put path as argument, but it wouldn’t be used - chosen transport wouldn’t utilize it.  Or some debug warning if temporary path is used.